mewarpx/mewarpx/diags_store/field_diagnostic.py

```diff
@@ -70,24 +84,65 @@ def plot_parameters(self):
+                            plot_name="rho_" + plot_name
+                        )
                 elif param == 'phi':
-                    data = np.array(mwxrun.get_gathered_phi_grid(include_ghosts=False)[0])
-
-                    fig, ax = plt.subplots(1, 1, figsize=(14, 14))
-                    plotting.ArrayPlot(
-                        array=data,
-                        template='phi', xaxis='z', yaxis='x', ax=ax,
-                        draw_image=True, draw_contourlines=False,
-                        cmap='viridis', plot_name="phi_" + plot_name
-                    )
+                    data = mwxrun.get_gathered_phi_grid(include_ghosts=False)
+                    if mwxrun.me == 0:
+                        data = np.array(data[0])
+
+                        fig, ax = plt.subplots(1, 1, figsize=(14, 14))
+                        plotting.ArrayPlot(
+                            array=data,
+                            template='phi', xaxis='z', yaxis='x', ax=ax,
+                            draw_image=True, draw_contourlines=False,
+                            cmap='viridis', plot_name="phi_" + plot_name
+                        )
                 else:
                     print("Attempted to plot a parameter for which plotting has not been implemented:", param)
+
+    def check_for_end_of_sim(self):
+        if mwxrun.get_it() == mwxrun.simulation.max_steps:
+            self.do_post_processing()
+
+    def do_post_processing(self):
+        if mwxrun.me == 0:
+            constants = picmi.constants
+            data_dirs = glob.glob('diags/diags*')
+
+            if len(data_dirs) == 0:
+                raise RuntimeError("No data files found.")
+
+            for datafolder in data_dirs:
+                if not "old" in datafolder:
+
+                    with pytest.warns(ResourceWarning):
+                        print('Reading ', datafolder, '\n')
+                        ds = yt.load( datafolder )
+
+                        grid_data = ds.covering_grid(
+                            level=0, left_edge=ds.domain_left_edge, dims=ds.domain_dimensions
+                        )
+
+                        for parameter in self.diag_data_list:
+                            plot_name = parameter + "_" + datafolder.replace("diags/diags", "")
+                            data = np.mean(
+                                grid_data[parameter].to_ndarray()[:,:,0], axis=0
+                            ) / constants.q_e
+
+                            fig, ax = plt.subplots(1, 1, figsize=(14, 14))
+                            template = "rho" if "rho" in parameter else "phi"
+                            plotting.ArrayPlot(
+                                    array=data,
+                                    template=template, xaxis='z', yaxis='x', ax=ax,
+                                    draw_image=True, draw_contourlines=False,
+                                    cmap='viridis', plot_name=plot_name
```

Comment:
Get rid of things that are now default.
```suggestion
                                    plot_name=plot_name
```